src/Components/Components/src/Rendering/Renderer.cs

```diff
@@ -383,7 +380,18 @@ private ComponentState GetOptionalComponentState(int componentId)
 
         private void ProcessRenderQueue()
         {
-            _isBatchInProgress = true;
+            lock (_asyncWorkLock)
```

Comment:
Would something like Interlocked.Exchange work better here? See the example here: https://docs.microsoft.com/en-us/dotnet/api/system.threading.interlocked.exchange?view=netframework-4.7.2#System_Threading_Interlocked_Exchange_System_Int32__System_Int32_

Reply:
I'm not too concerned about this lock being contested, and we already use it on other places.

Reply:
Right, but the sample uses a int value to do basically the same pattern that's being done here. Seems simpler to use the right tool here, no?

Reply:
It feels like premature optimization to me, I would rather have this represented as a bool than as an int for the sake of using a "faster" locking primitive. (Which we also don't know if it works in WebAssembly).